src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -127,6 +128,8 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
         }
     }
 
+    ov::pass::ConvertPrecision(ov::element::bf16, ov::element::f16).run_on_model(model);
```

Comment:
Could you please put a `FIXME` here to find the best place for this transformation? I'm not 100% sure it should be here and run unconditionally